hazelcast-client/src/test/java/com/hazelcast/client/starter/HazelcastClientStarter.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.client.starter;
+
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.test.starter.HazelcastAPIDelegatingClassloader;
+import com.hazelcast.test.starter.HazelcastStarter;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import static com.hazelcast.test.starter.HazelcastProxyFactory.proxyObjectForStarter;
+import static com.hazelcast.test.starter.HazelcastStarter.getConfig;
+import static com.hazelcast.test.starter.HazelcastStarter.getTargetVersionClassloader;
+import static com.hazelcast.test.starter.Utils.rethrow;
+import static java.lang.Thread.currentThread;
+
+public class HazelcastClientStarter {
+
+    private HazelcastClientStarter() {
+    }
+
+    public static HazelcastInstance newHazelcastClient(String version, boolean enterprise) {
+        return newHazelcastClient(version, null, enterprise);
+    }
+
+    public static HazelcastInstance newHazelcastClient(String version, ClientConfig clientConfig, boolean enterprise) {
```

Comment:
enterprise flag does not look very relevant for OSS code, does it?

Reply:
This is the infrastructure to test different versions of instances, and it is used by both open source and enterprise. We can abstract this part probably. This is not a new code by the way, I just moved it from `hazelcast` to `hazelcast-client` because it was not relevant there :).